R/tmb-methods.R

```diff
@@ -624,7 +624,9 @@ simulate.mmrm_tmb <- function(object,
         newtheta <- theta_chol %*% matrix(rnorm(ntheta), ncol = 1) + object$theta_est
         # Recalculate betas with sampled thetas.
         hold <- object$tmb_object$report(newtheta)
-        predict_res <- h_get_prediction(tmb_data, newtheta, hold$beta, hold$beta_vcov)
+        # Resample betas given new beta distribution.
+        beta_sample <- hold$beta + VCOV %*% rnorm(length(hold$beta))
```

Comment:
I might be fundamentally misunderstanding something here. Sampling $\beta_{sample}$ from the given distribution should be a very simple one-liner here. In fact, we have `beta_vcov` from `object$tmb_object$report()` that we could call directly, but we want to speed this up by returning $L$ and $D$ from the `C++` side. BUT, then $L$ and $D$ need to functions of the covariance of $\beta$, whereas if I'm reading right they're functions of $\theta$, no?

FYI: `VCOV` is just a placeholder, hence the canceled checks.

Reply:
So we have a 1:1 correspondence between beta_vcov and L&D. Both are functions of theta and we can modify the C++ side to also report L&D

Reply:
But I can also do this tomorrow myself no worries if it is not clear yet.